src/CSParametersRegistry.sol

```diff
@@ -238,10 +240,16 @@ contract CSParametersRegistry is
     /// @inheritdoc ICSParametersRegistry
     function setRewardShareData(
         uint256 curveId,
-        PivotsAndValues calldata data
+        KeyIndexValueInterval[] calldata data
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        _validatePivotsAndValues(data);
-        _rewardShareData[curveId] = data;
+        _validateKeysCountValueIntervals(data);
+        KeyIndexValueInterval[] storage intervals = _rewardShareData[curveId];
+        if (intervals.length > 0) {
+            delete _rewardShareData[curveId];
```

Comment:
Are you sure that this is needed and will yield a desired result? AFAIK delete will only nullify original storage slot, but not the underlining arrays

Reply:
We push new values, so we do need to `delete` (nullify the length) to "clean" the pointer. So, technically, there were old values, but they won't be reached because of the new length. And new values will overwrite the old ones.

Added tests for this case

Reply:
btw, we do the same when updating bond curve